docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,257 @@
+      by response header ``X-Trino-Started-Transaction-Id`` and
+      cleared by ``X-Trino-Clear-Transaction-Id``.
+  * - ``X-Trino-Client-Info``
+    - Contains arbitrary information about the client program
+      submitting the query.
+  * - ``X-Trino-Client-Tags``
+    - A comma-separated list of "tag" strings, used to identify
+      Trino resource groups.
+  * - ``X-Trino-Resource-Estimate``
+    - A comma-separated list of ``resource=value`` type
+      assigments.  The possible choices of ``resource`` are
+      "EXECUTION_TIME", "CPU_TIME",  "PEAK_MEMORY" and
+      "PEAK_TASK_MEMORY".  "EXECUTION_TIME" and "CPU_TIME"
+      have values specified as airlift ``Duration`` strings,
+      whose format is a double precision number followed by
+      a ``TimeUnit`` string, e.g., of ``s`` for seconds,
+      ``m`` for minutes, ``h`` for hours, etc.  "PEAK_MEMORY" and
+      "PEAK_TASK_MEMORY" are specified as as airlift ``DataSize`` strings,
+      whose format is an integer followed by ``B`` for bytes; ``kB`` for
+      kilobytes; ``mB`` for megabytes, ``gB`` for gigabytes, etc.
+  * - ``X-Trino-Extra-Credential``
+    - Provides extra credentials to the connector.  The header is
+      a name=value string that is saved in the session ``Identity``
+      object.  The name and value are only meaningful to the connector.
+
+Client response headers
+=======================
+This table lists the supported client response headers.  After receiving a
+response, a client must update the request headers that will be used in
+subsequent requests to be consistent with the response headers received.
+
+.. list-table:: Client response headers
+  :widths: 30, 50
+  :header-rows: 1
+
+  * - Header Name
+    - Description
+  * - ``X-Trino-Set-Catalog``
+    - Instructs the client to set the catalog that will be sent in the
+      ``X-Trino-Catalog`` request header in subsequent client requests.
+  * - ``X-Trino-Set-Schema``
+    - Instructs the client to set the schema that will be sent in the
+      ``X-Trino-Schema`` request header in subsequent client requests.
+  * - ``X-Trino-Set-Session``
+    - The value of the ``X-Trino-Set-Session`` response header is a
+      name=value string, representing session attributes that are
+      meaningful to the Trino engine or a connector. Instructs the client
+      to add that name=value string to the ``X-Trino-Session``
+      request header to be used in subsequent client requests.
+  * - ``X-Trino-Clear-Session``
+    - Instructs the client to remove the session property with the
+      whose name is the value of the ``X-Trino-Clear-Session`` header
+      from the comma-separated list of session properties that will be
+      sent in the ``X-Trino-Session`` header in subsequent client requests.
+  * - ``X-Trino-Set-Role``
+    - Instructs the client to set ``X-Trino-Role`` request header to the
+      catalog role give by the value of the ``X-Trino-Set-Role`` header
```

Comment:
given

?

Reply:
Changed to "catalog role supplied by the ``X-Trino-Set-Role`` header"